web/src/unread.js

```diff
@@ -275,15 +275,32 @@ class UnreadTopicCounter {
             }
 
             let stream_count = 0;
+            let stream_unmute_count = 0;
             for (const [topic, msgs] of per_stream_bucketer) {
                 const topic_count = msgs.size;
-                if (!user_topics.is_topic_muted(stream_id, topic)) {
+                if (
+                    !user_topics.is_topic_muted(stream_id, topic) &&
+                    !user_topics.is_topic_unmuted(stream_id, topic)
+                ) {
                     stream_count += topic_count;
+                } else if (user_topics.is_topic_unmuted(stream_id, topic)) {
+                    stream_unmute_count += topic_count;
                 }
             }
-            res.stream_count.set(stream_id, stream_count);
+
+            if (stream_unmute_count) {
+                res.stream_count.set(stream_id, {
+                    stream_count: stream_unmute_count,
+                    has_unmute_unread: true,
+                });
+            } else {
+                res.stream_count.set(stream_id, {stream_count, has_unmute_unread: false});
+            }
+
             if (!stream_data.is_muted(stream_id)) {
                 res.stream_unread_messages += stream_count;
+            } else {
+                res.stream_unread_messages += stream_unmute_count;
```

Comment:
This felt very messy; I've rewritten this logic to hopefully be more understandable.